lib/segment/src/segment_constructor/segment_builder.rs

```diff
@@ -455,10 +456,11 @@ impl SegmentBuilder {
         Ok(true)
     }
 
-    pub fn build(
+    pub fn build<R: Rng + ?Sized>(
```

Comment:
Usage of generics could make build times/binary size worse

Is it possible to use a non-generic type here? (e.g. `StdRng` or `&mut dyn RngCore`).

Reply:
From what I saw, we only use `ThreadRng` and `StdRng`. Do you know how much impact it would have if we switch to dyn?